sound/soc/sof/intel/hda.h

```diff
@@ -421,7 +421,7 @@ struct sof_intel_hda_stream {
 	struct snd_sof_dev *sdev;
 	struct hdac_ext_stream hda_stream;
 	struct sof_intel_stream stream;
-	int hw_params_upon_resume; /* set up hw_params upon resume */
+	int prepared; /* hw params set */
```

Comment:
@kv2019i just wondering if we can just use the link_prepared in hdac_ext_stream instead of this?

Reply:
@ranj063 Excellent point. I was considering that at first, but the resulting code looked a bit hard to follow (something that continues to bug me with our hdac/hdac_ext reuse) -- this is the DAI state after all that we are tracking.

But, but, when I grepped the code, 1) link_prepared is not really used for much, 2) it's not set correctly in all flows, and 3) it should be set/cleared in all the same places I was setting the new "prepared" field.

So net result with 1+2+3 combined, link_prepared management needs some fixes and after I do those, it can completely replace the SOF specific field. I did this now in updated patchset. The readability aspect is still a bit concerning, but I'll see if I can address that in a separate PR. I don't want to do this now as I'll then conflict with various other patches (including PR1072 Rander is doing).